src/core/acroform.cpp

```diff
@@ -0,0 +1,46 @@
+// SPDX-FileCopyrightText: 2022 James R. Barlow
+// SPDX-License-Identifier: MPL-2.0
+
+#include <qpdf/QPDF.hh>
+#include <qpdf/Constants.h>
+#include <qpdf/Types.h>
+#include <qpdf/DLL.h>
+#include <qpdf/QPDFExc.hh>
+
+#include <qpdf/QPDFObjectHandle.hh>
+#include <qpdf/QPDFAcroFormDocumentHelper.hh>
+
+#include <pybind11/pybind11.h>
+#include <pybind11/stl.h>
+
+#include "pikepdf.h"
+
+void init_acroform(py::module_ &m)
+{
+    py::class_<QPDFAcroFormDocumentHelper,
+        std::shared_ptr<QPDFAcroFormDocumentHelper>,
+        QPDFObjectHelper>(m, "AcroFormDocument")
+        .def(
+            py::init([](QPDF &q) {
+                QPDFAcroFormDocumentHelper afdh(q);
+
+                return afdh;
+            }),
+            py::keep_alive<0, 1>(), // LCOV_EXCL_LINE
+            py::arg("q")
+        )
+        .def(
+            "set_form_field_name",
```

Comment:
For this to be useful to anyone who say, wants to know what form fields are present before changing their names, you'll need to bind QPDFFormFieldObjectHelper as well. Then one should be able to retrieve form fields from the form and manipulate the field name from there.

I'm not going to accept a PR that solves your issue alone and isn't helpful to others. It needs to be of general use. All of the methods of QPDFFormFieldObjectHelper and QPDFAcroFormDocumentHelper should be exposed. You're probably going to need them anyway.

Please bind get/set functions as `def_property` which is equivalent to setting a `@property` with a pythonic getter and setter. Then one would use `form.fields[...].name = 'MyName'` etc.

Reply:
I added the method getFormFields, although it is mandatory for the usage that I solve. You could simplify use it as I showed in the test.
I dont know how I should do the binding with def_property.
If I understand you would me to add it to the binding of the class QPDFFormFieldObjectHelper but the method is on the class QPDFAcroFormDocumentHelper.

Implementing all the methods would take me too much time as I'm completely a novice in python and just discovered how pybind11 works, we could merge the base that I implemented and someone could improve it in the future